File: back/src/models/annonce.js
const {Sequelize, DataTypes} = require('sequelize');
const {sequelize} = require('./sqlite.db');
const annonce = sequelize.define('annonce',
    {
    ID_Annonce: {
      autoIncrement: true,
      type: DataTypes.BIGINT.UNSIGNED,
      allowNull: false,
      primaryKey: true
    },
    libelleAnnonce: {
      type: DataTypes.STRING(100),
      allowNull: false
    },
    descriptionAnnonce: {
      type: DataTypes.TEXT,
      allowNull: false
    },
    ID_Utilisateur: {
      type: DataTypes.INTEGER,
      allowNull: false
    },
    prix: {
      type: DataTypes.DECIMAL(11,2),
      allowNull: false,
      defaultValue: 0.00
    },
    PossibleNego: {
      type: DataTypes.BOOLEAN,
      allowNull: false,
      defaultValue: 0
    },
    ID_Categorie: {
      type: DataTypes.INTEGER,
      allowNull: false
    }
  }, {
    sequelize,
    tableName: 'annonce',
    timestamps: false,
    indexes: [
      {
        name: "pk_ID_Annonce",
        unique: true,
        using: "BTREE",
        fields: [
          { name: "ID_Annonce" },
        ]
      },
      {
        name: "fk_ID_Utilisateur_Annonce",
        using: "BTREE",
        fields: [
          { name: "ID_Utilisateur" },
        ]
      },
      {
        name: "fk_ID_Categorie_Annonce",
        using: "BTREE",
        fields: [
          { name: "ID_Categorie" },
        ]
      },
    ]
})
module.exports = annonce

File: front mobile/app/src/main/java/com/example/leboncoup/Annonce.kt
package com.example.leboncoup

import com.google.gson.annotations.SerializedName

class Annonce {

    @SerializedName("ID_Annonce")
    var ID_Annonce : Int? = null

    @SerializedName("libelleAnnonce")
    var libelleAnnonce : String? = null

    @SerializedName("descriptionAnnonce")
    var descriptionAnnonce : String? = null

    @SerializedName("prix")
    var prix : Int? = null

    @SerializedName("PossibleNego")
    var PossibleNego : Boolean? = null

    @SerializedName("ID_Categorie")
    var ID_Categorie : Int? = null

    override fun toString(): String {
        return "$ID_Annonce, $libelleAnnonce, $descriptionAnnonce, $prix, $PossibleNego, $ID_Categorie"
    }
}